src/convert.cpp

```diff
@@ -0,0 +1,346 @@
+	//cout << "still works" << endl;
+	//nodes information
+
+		
+	while(!ifs.eof()){
+		getline(ifs,line);
+		if(regex_match(line,pattern)){
+			int check = 0;
+			int target_start =0;
+			unsigned int i = 0;
+			while(i < line.length())
+			{
+				if(isspace(line[i]) && check == 0){
+					check = 1;
+					string start =  line.substr(0,i);
+					ofs<<start<<" ";
+				}
+				if(!isspace(line[i]) && check == 1){
+					check = 2;
+					target_start = i;
+				}
+				if(isspace(line[i]) && check == 2){
+					check = 0;
+					string target = line.substr(target_start,i-target_start);
+					ofs<<target<< endl;
+						break;
+				}
+				i++;
+			}
+			getline(ifs,line);
+		}
+	}
+	
+	ifs.close();
+	ofs.close();
+}
+
+
+
+void convert_grapgml(string inp)
+{
+
+    regex source("(.*)(source)(.*)");
+    regex target("(.*)(target)(.*)");
+    ifstream inFile;
+    ofstream myfile;
+    inFile.open(inp);
+    myfile.open (" tmpel");
+    if (!inFile) {
+        cout << "Unable to open file";
+        exit(1); // terminate with error
+    }
+    string line;
+    string sub;
+    while (inFile.good()) {
+
+	getline (inFile,line);
+	if(regex_match(line,source))
+{
+		size_t  s_place = line.find("source=");
+		string re = line.substr(s_place);
+		int n = 7;
+		if(re[7] == '"')
+			n++;
+		string sub = re.substr(n);
+		char f = ' ';
+                if(n == 8)
+			f = '"';
+		string final_source;
+		for(string::iterator it=sub.begin(); it!=sub.end(); ++it)
+		{
+			if(*it == f)
+				break;
+			final_source.push_back(*it);
+			
+		}
+		size_t  t_place = re.find("target=");
+		string t_re = re.substr(t_place);
+		int x = 7;
+		if(t_re[7] == '"')
+			x++;
+		string t_sub = t_re.substr(n);
+		string final_target;
+		for(string::iterator tit=t_sub.begin(); tit!=t_sub.end(); ++tit)
+		{
+			if(*tit == f || *tit == '>'||*tit == '/' )
+				break;
+			final_target.push_back(*tit);
+			
+		}
+		myfile << final_source << "  " << final_target << "\n";
+    }
+}
+    myfile.close();
+    inFile.close();
+
+}
+void convert_gml(string inp)
+{    
+    clock_t t;
+    t = clock();
+    regex source("(.*)(source)(.*)");
+    regex target("(.*)(target)(.*)");
+    ifstream inFile;
+    ofstream myfile;
+    inFile.open(inp);
+    myfile.open (" tmpel");
+	t = clock() - t;
+    cout << "open file needs: " <<  t << endl;
+    clock_t t1;
+    t1 = clock();
+    if (!inFile) {
+        cout << "Unable to open file";
+        exit(1); // terminate with error
+    }
+    string line;
+    string sub;
+    while (inFile.good()) {
+
+	getline (inFile,line);
+	if(regex_match(line,source))
+		
+		myfile << line.substr(9) << ' ';
+	if(regex_match(line,target))
+		myfile << line.substr(9) << "\n";
+continue;
+	
+
+
+    }
+	t1 = clock() - t1;
+     cout << "write file needs: " << t1 << endl;
+    myfile.close();
+    inFile.close();
+
+}
+
+
+
+
+void convert_csv( string filename){
+	ifstream ifs;
+	ifs.open(filename, ifstream::in);
+	ofstream ofs;
+	ofs.open(" tmpel");
```

Comment:
In `ofs.open(" tmpel");`
Extra space in the name, also more importantly hard coding the file name is not good idea, if we run multiple sana's they would try to write to same file and fail. 

Should probably be a combination of a random number + the original filename.

Reply:
Yes I suggest using getpid()